### PR TITLE
Add Navigation Component

### LINKED ISSUE #31: Create Navigation Component
Create navigation component that can be used across mobile/desktop views.

Each icon carries the following functionality:
- Go to "Add Item" view
- Go to "List" view
- Sign out of current list and go to homepage
- Share token
     - Text is automatically copied to clipboard

### PR DESCRIPTION
**Pull request by sanrafa**
## Description

This PR adds a Navigation component containing the app logo and MaterialUI icon buttons with the following functionality:
1. Navigate to "Add Item" view
2. Navigate to "List" view
3. Sign out and redirect to homepage
4. Copy and paste list token for sharing

I decided to incorporate the Navigation component with the common PageWrapper component by passing it as a prop. I did this for styling purposes, as it made it easier to remain consistent with the responsive styling of the wrapper.

## Related Issue

Closes #31 

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |
|  ✓   | :link: Update dependencies |

### Before

navigation-before

### After

Desktop version:
navigation-add-desktop

Mobile version:
navigation-add-mobile

### CODE REVIEW

File: src/components/Navigation/Navigation.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    navigator.clipboard.writeText(token).then(() =>
+      toast('Token copied to clipboard!', {
+        autoClose: 3000,
+        hideProgressBar: true,
+        pauseOnHover: false,
+        draggable: false,
+      }),
+    );
+  };
+
```
Comment (danainjax) [inline src/components/Navigation/Navigation.jsx]: For future development, I wonder if we can also display the token? Modal? Toast?

File: src/components/Navigation/Navigation.css [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    display: flex;
+    justify-content: center;
+  }
+}
+
+.logo > img {
+  max-width: 80%;
+  height: auto;
+  padding: 1rem;
+}
```
Comment (danainjax) [inline src/components/Navigation/Navigation.css:23]: Is there a way on hover to see what the nav icon will do? When I hover on the logout icon, can it display "logout"? Not anything preventing this from being approved, just an idea for user experience.

**Review by danainjax — APPROVED** [review]
This is looking fantastic! I left two suggestions for future development/discussion.

File: src/components/Navigation/Navigation.jsx (hunk 1 above)
Comment (sanrafa) [reply src/components/Navigation/Navigation.jsx]: For clarification, display it when copied or displayed consistently on the page? I think it's a great suggestion, though - it adds more transparency for the user!

File: src/components/Navigation/Navigation.css (hunk 2 above)
Comment (sanrafa) [reply src/components/Navigation/Navigation.css:23]: Hmm, I'm not sure if MUI has built-in methods for this but I'll do some research and add it in - might be able to do it using some kind of tooltip. I tried to use the clearest icons possible, but I agree - I hate when sites throw me a bunch of icons and I have to figure out what they do by clicking them.

File: src/components/Navigation/Navigation.jsx (hunk 1 above)
Comment (danainjax) [reply src/components/Navigation/Navigation.jsx]: When copied. What do you think?

File: src/components/Navigation/Navigation.css (hunk 2 above)
Comment (danainjax) [reply src/components/Navigation/Navigation.css:23]: Agree also. Example: Amazon. Lots of icons. Would not know what it does until you click. I think is good as is. This is our first pass and we're on the final week.

File: src/components/Navigation/Navigation.jsx (hunk 1 above)
Comment (sanrafa) [reply src/components/Navigation/Navigation.jsx]: Figured out a way to add it to the toast notification. It was fun figuring it out, so thank you for the suggestion!

File: src/components/Navigation/Navigation.css (hunk 2 above)
Comment (sanrafa) [reply src/components/Navigation/Navigation.css:23]: MaterialUI made it surprisingly easy to add in, so I went ahead and implemented it. Thank you :)

File: src/components/Navigation/Navigation.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        {
+          autoClose: 5000,
+          hideProgressBar: true,
+          pauseOnHover: true,
+          draggable: false,
+          closeOnClick: false,
+        },
+      ),
+    );
+  };
```
Comment (vividavila98) [inline src/components/Navigation/Navigation.jsx:39]: I didn't know about `navigator.clipboard.writetext` that's super useful!

File: src/components/Navigation/Navigation.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      <Tooltip title="Sign out">
+        <IconButton aria-label="sign out" onClick={() => handleSignOut()}>
+          <LogoutIcon sx={navIconStyle} />
+        </IconButton>
+      </Tooltip>
+      <Tooltip title="Share token">
+        <IconButton aria-label="share token" onClick={() => copyToClipboard()}>
+          <ShareIcon sx={navIconStyle} />
+        </IconButton>
+      </Tooltip>
```
Comment (vividavila98) [inline src/components/Navigation/Navigation.jsx:71]: Good call on adding the tooltips!

File: src/components/Navigation/Navigation.jsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          closeOnClick: false,
+        },
+      ),
+    );
+  };
+
+  const navIconStyle = {
+    fontSize: '2.5rem',
+    display: 'inline-block',
+  };
```
Comment (vividavila98) [inline src/components/Navigation/Navigation.jsx:45]: Would it be possible for the icons to be black instead of gray?

**Review by vividavila98 — APPROVED** [review]
I learned so much reviewing this! Great job Rafa :))

File: src/components/Navigation/Navigation.jsx (hunk 5 above)
Comment (sanrafa) [reply src/components/Navigation/Navigation.jsx:45]: Good call, just made that change on the final commit :)

File: src/components/Navigation/Navigation.jsx (hunk 1 above)
Comment (danainjax) [reply src/components/Navigation/Navigation.jsx]: This is exciting!

File: src/components/Navigation/Navigation.jsx (hunk 4 above)
Comment (danainjax) [reply src/components/Navigation/Navigation.jsx:71]: Tooltips are cool!

File: src/components/Navigation/Navigation.jsx (hunk 3 above)
Comment (danainjax) [reply src/components/Navigation/Navigation.jsx:39]: Who doesn't love a token/toast? This is a great enhancement. Thank you for developing it, Rafa.

**Pull request merged**